<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <style>
    body {
      background: #f4f8fc;
      margin: 0;
      font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    }
    h2 {
      color: #003366;
    }
    table {
      width: 100%;
      border-collapse: separate;
      border-spacing: 0 12px;
    }
    th, td {
      padding: 8px 15px;
      text-align: left;
    }
    th {
      background-color: #f0f6ff;
      border-bottom: 2px solid #cce0f5;
    }
    td {
      color: #2a3b55;
    }
    a {
      color: #0066cc;
      text-decoration: none;
    }
  </style>
</head>
<body>


<div style="
  background: linear-gradient(90deg, #dfeffc, #e6f2ff);
  color: #003366;
  padding: 20px 30px;
  border-radius: 12px;
  border-left: 6px solid #0066cc;
  border-right: 6px solid #0066cc;
  box-shadow: 0 4px 12px rgba(0, 102, 204, 0.15);
  text-align: center;
  font-weight: bold;
  font-size: 35px;
  letter-spacing: 1px;
  margin: 40px auto 30px auto;
  width: fit-content;
">
🔹02.📥 Data Collection :
</div>

<div style="
  max-width: 1100px;
  margin: 0 auto 40px auto;
  background: #f9fbfe;
  border-radius: 12px;
  padding: 30px;
  box-shadow: 0 0 15px rgba(0, 102, 204, 0.1);
">
<div style="background-color: #e6f2ff; 
            #border-left: 6px solid #005b96; 
            padding: 12px 20px; 
            margin: 20px 0; 
            border-radius: 10px; 
            box-shadow: 0 2px 5px rgba(0,0,0,0.1); 
            font-family: 'Segoe UI', sans-serif; 
            font-size: 25px; 
            font-weight: 600; 
            color: #003366;">
  2.1 📌 About The Dataset:
</div>


  

  <p>
    The <strong>Bank Customer Churn</strong> dataset is a commonly used dataset for predicting customer churn in the banking industry and we get it from Kaggle.<br>
    🔗 <a href="https://www.kaggle.com/datasets/shubhammeshram579/bank-customer-churn-prediction" target="_blank">
      View Dataset on Kaggle
    </a>
  </p>
  <p>
    It contains information on bank customers who either <strong>left the bank</strong> or <strong>continue to be customers</strong>.
  </p>


  <div style="background-color: #e6f2ff; 
            #border-left: 6px solid #005b96; 
            padding: 12px 20px; 
            margin: 20px 0; 
            border-radius: 10px; 
            box-shadow: 0 2px 5px rgba(0,0,0,0.1); 
            font-family: 'Segoe UI', sans-serif; 
            font-size: 25px; 
            font-weight: 600; 
            color: #003366;">
  2.2 📌 Meta Data (Column Descriptions):
</div>


  <p>This section provides detailed descriptions for each column in the dataset. Understanding these features is essential before performing any kind of analysis or modeling.</p>

  <table>
    <thead>
      <tr>
        <th>📌 Column Name</th>
        <th>📝 Description</th>
      </tr>
    </thead>
    <tbody>
      <tr><td>RowNumber</td><td>A unique identifier for each record.</td></tr>
      <tr><td>Customer ID</td><td>A unique identifier for each customer.</td></tr>
      <tr><td>Surname</td><td>The customer's surname or last name.</td></tr>
      <tr><td>Credit Score</td><td>A numerical value representing the customer's credit score.</td></tr>
      <tr><td>Geography</td><td>The country where the customer resides (France, Spain, or Germany).</td></tr>
      <tr><td>Gender</td><td>The customer's gender (Male or Female).</td></tr>
      <tr><td>Age</td><td>The customer's age.</td></tr>
      <tr><td>Tenure</td><td>The number of years the customer has been with the bank.</td></tr>
      <tr><td>Balance</td><td>The customer's account balance.</td></tr>
      <tr><td>NumOfProducts</td><td>The number of bank products the customer uses (e.g., savings account, credit card).</td></tr>
      <tr><td>HasCrCard</td><td>Whether the customer has a credit card (1 = yes, 0 = no).</td></tr>
      <tr><td>IsActiveMember</td><td>Whether the customer is an active member (1 = yes, 0 = no).</td></tr>
      <tr><td>EstimatedSalary</td><td>The estimated salary of the customer.</td></tr>
      <tr><td>Exited</td><td>Whether the customer has churned (1 = yes, 0 = no).</td></tr>
    </tbody>
  </table>

</div>

</body>
</html>


<div style="background-color: #e6f2ff; 
            #border-left: 6px solid #005b96; 
            padding: 12px 20px; 
            margin: 20px 0; 
            border-radius: 10px; 
            box-shadow: 0 2px 5px rgba(0,0,0,0.1); 
            font-family: 'Segoe UI', sans-serif; 
            font-size: 25px; 
            font-weight: 600; 
            color: #003366;">
  2.3 📌 Import Libraries:
</div>


In [9]:
# ------------------ 1. 📦 Custom Utilities ------------------
import MyLibrary as lib  # My custom helper functions (e.g., readfile)

# ------------------ 2. 📊 Data Handling ------------------
import pandas as pd
import numpy as np
import os

<div style="background-color: #fff3e0;
            #border-left: 5px solid #fb8c00;
            margin: 20px 30px;
            padding: 16px 24px;
            border-radius: 8px;
            font-family: 'Segoe UI', sans-serif;
            font-size: 18px;
            font-weight: 500;
            color: #e65100;">
  ✅ <strong>Conclusion:</strong> All essential Python libraries were successfully imported.<br><br>
  ▪️ <strong>📦 Custom Utilities:</strong> <code>MyLibrary</code> – Local helper functions (e.g., <code>readfile()</code>)<br>
  ▪️ <strong>📊 Data Handling:</strong> <code>pandas</code>, <code>numpy</code>, <code>os</code> – For data structures, operations, and file access<br><br>
  These imports form the foundation for loading, manipulating, and accessing dataset files throughout the notebook.
</div>

<div style="background-color: #e6f2ff; 
            #border-left: 6px solid #005b96; 
            padding: 12px 20px; 
            margin: 20px 0; 
            border-radius: 10px; 
            box-shadow: 0 2px 5px rgba(0,0,0,0.1); 
            font-family: 'Segoe UI', sans-serif; 
            font-size: 25px; 
            font-weight: 600; 
            color: #003366;">
  2.4 📌 Reading Dataset:
</div>


In [12]:
# I Will import function of readfile() from My Local own Library to read any file like csv file, excell file or etc.
data = lib.readfile("../Data/bankcustomer.csv")


Dataset of '../Data/bankcustomer.csv' loaded successfully.

Dataset Shape is : 10002 rows × 14 columns

Dataset Columns are :
 ['rownumber', 'customerid', 'surname', 'creditscore', 'geography', 'gender', 'age', 'tenure', 'balance', 'numofproducts', 'hascrcard', 'isactivemember', 'estimatedsalary', 'exited'] 




<div style="background-color: #fff3e0;
            #border-left: 5px solid #fb8c00;
            margin: 20px 30px;
            padding: 16px 24px;
            border-radius: 8px;
            font-family: 'Segoe UI', sans-serif;
            font-size: 18px;
            font-weight: 500;
            color: #e65100;">
  ✅ <strong>Conclusion:</strong> Dataset loaded successfully.<br><br>
  ▪️ File path: <code>../Data/bankcustomer.csv</code><br>
  ▪️ Dataset shape: <code>10002 rows × 14 columns</code><br>
  ▪️ Columns:<br>
  <code>
  ['rownumber', 'customerid', 'surname', 'creditscore', 'geography', 'gender', 'age', 'tenure',<br>
   'balance', 'numofproducts', 'hascrcard', 'isactivemember', 'estimatedsalary', 'exited']
  </code>
</div>
